OnlineStore/OnlineStore.Data/Repositories/Implementations/ProductRepository.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using OnlineStore.Data.Contexts;
+using OnlineStore.Data.Entities;
+using OnlineStore.Data.Interfaces;
+
+namespace OnlineStore.Data.Repositories.Implementations
+{
+    internal class ProductRepository : Repository<Product>, IProductRepository
+    {
+        public ProductRepository(ProductsOrdersContext productsOrdersContext)
+            : base(productsOrdersContext)
+        {
+
+        }
+
+        public async Task AddCategoriesToProductAsync(Int32 productId, ICollection<Int32> categoriesId)
+        {
+            var product = await DbSet.Include(p => p.Categories)
```

Comment:
This method needs to be refactored, as it can be simplified. Also, checks and exception handling should be in the services, not here